Lesson6/main.js

```diff
@@ -0,0 +1,25 @@
+const ul = document.getElementById("js-ul");
+const fragment = new DocumentFragment();
+const features = [{to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"},
+{to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}];
+
+const checkFeatures = new Promise((resolve) => {
+  setTimeout(() => {
+    resolve(features);
+  }, 3000);
+}).then((value)　=> {
```

Comment:
I agree with Chihiro-san.
If you want to assign to a variable, I think you should delimit the process.

```suggestion
  setTimeout(() => {
    resolve(features);
  }, 3000);
});

checkFeatures.then((value)　=> {
```

Reply:
Thank you for sharing the way to check how long it takes to execute code comes after setTime out. 
I did not included it in my latest change, but I played it around to see how it works.